content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesAction.java

```diff
@@ -10521,5 +10397,229 @@ public void doViewTrash(RunData data)
+		HttpServletResponse response = (HttpServletResponse)threadLocalManager.get(RequestFilter.CURRENT_HTTP_RESPONSE);
+
+		List<String> selectedFolderIds = new ArrayList<>();
+		List<String> selectedFiles = new ArrayList<>();
+		for (ListItem listItem : zipDownloadItems) {
+			if (listItem.isCollection()) {
+				selectedFolderIds.add(listItem.getId());
+			} else {
+				selectedFiles.add(listItem.getId());
+			}
+		}
+
+		if ("imscc".equals(exportType)) {
+			String collectionId = (String) state.getAttribute(STATE_COLLECTION_ID);
+			Reference ref = EntityManager.newReference(ContentHostingService.getReference(collectionId));
+			String siteId = ref.getContext();
+			try {
+				CCExport ccExport = new org.sakaiproject.exporter.util.CCExport(siteId, ComponentManager.get(org.sakaiproject.content.api.ContentHostingService.class));
+				ccExport.doExport(selectedFolderIds, selectedFiles, response);
+			} catch (IdUnusedException ide) {
+				// Shouldn't happen by this stage.
+				logger.warn("IdInvalidException site ID:" + siteId + " collection ID: " + collectionId);
+			}
+		}
+		else if ("zip".equals(exportType)) {
+				// Use the site title for the zip name, remove spaces though.
+				String siteTitle = (String) state.getAttribute(STATE_SITE_TITLE);
+				siteTitle = siteTitle.replace(" ", "");
+				new ZipContentUtil().compressSelectedResources((String)state.getAttribute(STATE_SITE_ID), siteTitle, selectedFolderIds, selectedFiles, response);
+		}
+
+		if (state.getAttribute(STATE_MESSAGE) == null)
+		{
+			state.setAttribute (STATE_MODE, MODE_LIST);
+			state.removeAttribute(STATE_EXPORTDOWNLOAD_SET);
+
+			if (state.getAttribute (STATE_SELECT_ALL_FLAG).equals(Boolean.TRUE.toString()))
+			{
+				state.setAttribute (STATE_SELECT_ALL_FLAG, Boolean.FALSE.toString());
+			}
+		}
+	}
+
+	private long getCollectionRecursiveSize(ContentCollection currentCollection, long maxIndividualFileSize, long zipMaxTotalSize, Set<String> zipSingleFileSizeExceeded) throws ZipMaxTotalSizeException
+	{
+		long total=0;
+
+		List items = currentCollection.getMemberResources();
+		Iterator it = items.iterator();
+		while(it.hasNext())
+		{
+			ContentEntity myElement = (ContentEntity) it.next();
+			if (myElement.isResource()) 
+			{
+				long tempSize = ((ContentResource)myElement).getContentLength();
+				if (tempSize > maxIndividualFileSize) {
+					// Work out the file path without the site ID.
+					String filePath = myElement.getId().replace("/group/" + ToolManager.getCurrentPlacement().getContext(), "");
+					zipSingleFileSizeExceeded.add(filePath);
+				}
+				else {total=total+tempSize;}
+			}
+			else if (myElement.isCollection())
+			{
+				long tempSize = getCollectionRecursiveSize((ContentCollection)myElement, maxIndividualFileSize, zipMaxTotalSize, zipSingleFileSizeExceeded);
+				if (tempSize > zipMaxTotalSize) {
+					throw new ZipMaxTotalSizeException();
+				}
+				else {total=total+tempSize;}
```

Comment:
Sakai normally splits this out onto multiple lines.

Reply:
Have changed this.